com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/MaxRectsLayoutStrategy.java

```diff
@@ -607,34 +607,57 @@ private int contactPointScoreNode (int x, int y, int width, int height) {
 
-            for (int i = 0; i < freeRectangles.size(); i++) {
-                // Try to place the rectangle in upright (non-rotated) orientation.
-                RectNode currentNode = freeRectangles.get(i);
-                if (currentNode.rect.getWidth() >= width && currentNode.rect.getHeight() >= height) {
-                    int score = contactPointScoreNode(currentNode.rect.getX(), currentNode.rect.getY(), width, height);
-                    if (score > bestNode.score1) {
-                        bestNode.rect = new Rect(currentNode.rect.getId(), currentNode.rect.getIndex(), currentNode.rect.getX(), currentNode.rect.getY(), width, height);
-                        bestNode.score1 = score;
-                    }
+            // Try to place the rectangle in upright (non-rotated) orientation.
+            if (currRect.getWidth() >= width && currRect.getHeight() >= height) {
+                int score = contactPointScoreNode(currRect.getX(), currRect.getY(), width, height);
+                if (score > bestNode.score1) {
+                    bestNode.rect = new Rect(currRect.getId(), currRect.getIndex(), currRect.getX(), currRect.getY(), width, height);
+                    bestNode.score1 = score;
                 }
-                if (rotate && currentNode.rect.getWidth() >= rotatedWidth && currentNode.rect.getHeight() >= rotatedHeight) {
-                    // This was width,height -- bug fixed?
-                    int score = contactPointScoreNode(currentNode.rect.getX(), currentNode.rect.getY(), rotatedWidth, rotatedHeight);
-                    if (score > bestNode.score1) {
-                        bestNode.rect = new Rect(currentNode.rect.getId(), currentNode.rect.getIndex(), currentNode.rect.getX(), currentNode.rect.getY(), rotatedWidth, rotatedHeight);
-                        bestNode.score1 = score;
-                        bestNode.rect.setRotated(true);
-                    }
+            }
+
+            if (rotate && currRect.getWidth() >= rotatedWidth && currRect.getHeight() >= rotatedHeight) {
+            	// This was width,height -- bug fixed?
+                int score = contactPointScoreNode(currRect.getX(), currRect.getY(), rotatedWidth, rotatedHeight);
+                if (score > bestNode.score1) {
+                    bestNode.rect = new Rect(currRect.getId(), currRect.getIndex(), currRect.getX(), currRect.getY(), rotatedWidth, rotatedHeight);
+                    bestNode.score1 = score;
+                    bestNode.rect.setRotated(true);
                 }
             }
+
             return bestNode;
         }
 
+        private RectNode findPositionForNewNodeContactPoint(int width, int height, int rotatedWidth, int rotatedHeight, boolean rotate) {
+            RectNode globalBestNode = new RectNode();
+            globalBestNode.rect = new Rect(null, 0,0,0,0,0);
+            globalBestNode.score1 = Integer.MIN_VALUE; // maximize contact points
+
+            if (freeRectangles.size() < 100) {
```

Comment:
Not sure if it's a good idea, But I think there is no sense to run it in parallel for a small set of images. On the other hand, it is extra complexity in code... What do you think?

Reply:
Is there a drawback to running it in parallel also for a small set of images? If there isn't then I suggest doing it also for a small set.

Reply:
Sure, but surely not that much time? Packing an atlas with 1-100 images should be instant even with the overhead. But lets go ahead and merge. It's not a big deal!